pkg/controller/vitesskeyspace/vitesskeyspace_controller.go

```diff
@@ -179,6 +184,19 @@ func (r *ReconcileVitessKeyspace) Reconcile(request reconcile.Request) (reconcil
 	topoResult, err := r.reconcileTopology(ctx, vtk)
 	resultBuilder.Merge(topoResult, err)
 
+	wr, err := newWrangler(ctx, vtk.Spec.GlobalLockserver)
```

Comment:
It seems like you continue on even if we fail to make a wrangler. It makes sense that we don't abort the whole Reconcile just because we can't get resharding status, but we should avoid trying to use the nonexistent  wrangler. It'll probably be clearer to fix this and the Close() problem if we make a subroutine like `r.reconcileResharding`. It would also be nice to put that and all resharding-related code in a new file, `reconcile_resharding.go`.